### ISSUE TITLE
Hero Units Should Have Traits (if They Are Not a Unique Unit?)

### ISSUE THREAD

**Issue by Ashmyr** [wesnoth/wesnoth#7129]
### Describe the desired feature

In general, I think this should apply to all campaigns. If your leader is a regular unit and he doesn't have traits, his own recruits are better than him. Leaders that are unique units have their own stats, but I still think they should have traits. Traits add character to the characters (lol), as well.
I'm sure there might be more (such as allies or loyal units you gain along the way), but just looking at the code for the leaders for each campaign...
- Kai Krellis in Dead Water has no traits.
- Malin Keshar in Descent into Darkness has no traits, but is a unique unit.
- Gweddry in Eastern Invasion has no traits, but is a unit that he cannot recruit. However, I still think he should have them.
- Konrad and Li'sar in Heir to the Throne have no unique traits, but are both unique units.
- Baldras in Liberty has no traits.
- Rugnur in Sceptre of Fire has no traits.
- Kapou'e in Son of the Black-Eye has no traits.
- Haldric in The Rise of Wesnoth has no traits, but is a unique unit.
- Deoran in The South Guard has no traits, but is a unique unit.
- Arvith in A Tale of Two Brothers has no traits.
- Kaleh in Under the Burning Suns has no traits, but is a unique unit.

**Comment by Ashmyr** [comment]
I am also a fan of giving "illegal traits", ie, Jessene having _Dextrous_, and I feel that can add characterization as well. I think allied leaders (except for cutscenes) and Loyal Units should all have traits, but enemies don't have to.

**Comment by knyghtmare** [comment]
List of proposed traits for each and accompanying logic?

**Comment by Ashmyr** [comment]
I'm not quite sure for some. I think it needs to fit the personality and the campaign. If the leader has a lot of go to missions, maybe Quick? If the leader is expected to be a frontline unit, maybe Strong, Fearless or Resilient? Dextrous could help out leaders with a weak secondary attack and give them a small boost...

**Comment by CelticMinstrel** [comment]
Yes, if it's a unique unit you can make the trait be nothing but "sugar" by adjusting the base stats so that the trait brings them back to the original level.

If it's not a unique unit, you'd need to consider balance in choosing traits.

**Comment by Ashmyr** [comment]
Thoughts on leader traits:
Arvith: Strong, Intelligent
Deoran: Strong, Intelligent
Jessene: Dextrous, Strong
Gweddry: Resilient, Fearless
Konrad: Resilient, Dextrous
Baldras: Healthy, Fearless
Rugnur: Healthy, Quick
Li'sar: Strong, Fearless

I'm not sure about others for now, and I'm sure there's some hero units or allies that I've missed and should help brainstorm (Like, I assume Baran and Hoban from AtotB have no traits?)

**Comment by stevecotton** [comment]
For the unique units, nerfing stats and then giving them traits will mean their in-game stats no longer match the "Unit Description" encyclopedia page. That seems to be introducing a bug.

Giving the `fearless` trait changes the gameplay a bit.

> If your leader is a regular unit and he doesn't have traits, his own recruits are better than him.

Well, recruiting people who are better suited to the tasks at hand than yourself is a form of real-world leadership.

**Comment by Ashmyr** [comment]
Fair I guess, but I think your leader still should have some to show off aspects of his personality and stuff.

As for Fearless, I'm aware, and I have the most narrow criteria for it. Li'sar is the exception, not the rule (might make her too tanky, but you get her late in the campaign and she is the daughter of a terrible tyrannical queen). It'd mostly be for weaker units, or units that are at a disadvantage (Most of EI is versus Undead, and Gweddry has blade, which isn't effective, so it aids him out.)

**Comment by Ashmyr** [comment]
Also, I said before that I was unsure if you need units should have them because they are a unique unit. They already have pretty unique stats. However, I think all leaders who are default units should have traits, even if there is real world people who are worse at tasks than the people working under them, I feel like it would just kind of be a little impractical to not give them traits. I think most players want their leader unit to be a hero fighting on the front lines or some sort of unique or cool unit. If you have him just be a default unit with no traits, then that kind of ruins the immersion for me because his units are just objectively better than him. That's no fun at all when you notice it.

**Comment by CelticMinstrel** [comment]
> Giving the `fearless` trait changes the gameplay a bit.

In fact, it's equivalent to change the alignment to neutral, right? I don't think we should use it for leader units unless they are a standard unit type and it thematically makes sense. If it's a unique unit type (eg Li'sar) I don't think we should give the fearless trait.

**Comment by cooljeanius** [comment]
> 
>     * Kapou'e in Son of the Black-Eye has no traits.
> 

FWIW, I gave him Fearless and Intelligent in SotBEEE; see cooljeanius/Son_Of_The_Black_Eye_Easy_Edition@72909e7 and cooljeanius/Son_Of_The_Black_Eye_Easy_Edition#7

> Thoughts on leader traits: Arvith: Strong, Intelligent 

I don't think Intelligent really makes much sense for Arvith; my impression was that Baran was supposed to be more the brains of the family, while Arvith was the brawn. Regardless, the fact that he needs traits is a good one; I'll try to add some in my version...

**Comment by cooljeanius** [comment]
> (Like, I assume Baran and Hoban from AtotB have no traits?)

I just checked these two; Baran has Loyal and Intelligent, but Hoban doesn't have anything

**Comment by Slayer95** [comment]
> For the unique units, nerfing stats and then giving them traits will mean their in-game stats no longer match the "Unit Description" encyclopedia page. That seems to be introducing a bug.

Then what about giving them the traits in-name only? That is, keeping their base and final stats as is, and just let it say "strong". Then people could argue that the in-game help lists Lord unit as having 9-3 bow attack, and Konrad being dextrous means that it should have 10-3, but that not being the case would be a bug.

So would both behaviors be bug? I don't think so. IMO, the Lord unit being unique is an implementation detail. Wesnoth players needn't know that such unit is only intended to be used by HttT's Konrad. Therefore, player expectation is that a dextrous Lord should be different from a Lord and should, in fact, exist.
